ads/criteo.js

```diff
@@ -53,8 +59,60 @@ export function criteo(global, data) {
 /**
  * @param {!Window} global
  * @param {!Object} data
+ * @param {!function(*)} resultCallback
  */
-function setTargeting(global, data) {
+function criteoStandalone(global, data, resultCallback) {
+  const adUnitParams = {
+    'integrationmode': 'amp',
+    'placements': [
+      {
+        'slotid': data.slot,
+        'zoneid': data.zone,
+      },
+    ],
+  };
+  Criteo.RequestBids(adUnitParams, bids => {
+    listenForCreativeRequests();
+    for (const i in bids) {
+      const bid = bids[i];
+      if (data.adserver === 'DFP') {
+        const lir = data.lineItemRanges;
+        const targeting = Criteo.ComputeStandaloneDFPTargeting(bid, lir);
+        setTargeting(global, data, targeting);
+      }
+    }
+    resultCallback(null);
+  }, data.timeout);
+}
+
+function listenForCreativeRequests() {
```

Comment:
how likely you can move more logic into your `https://static.criteo.net/js/ld/publishertag.js`?

so that 
1) for your own benefit: any future change / bug fix does not depend on AMP release cycle
2) to not bloat AMP's binary size.